### PR TITLE
Make sure that a hidden field/question is never required

### LINKED ISSUE #7653: [CAL-1218] Can't book an event with hidden but required field.
Can't go through with a booking which has an hidden field which is marked required.

Thanks @Udit-takkar for reporting the issue.

<sub>[CAL-1218](https://linear.app/calcom/issue/CAL-1218/cant-book-an-event-with-hidden-but-required-field)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes #7653

**Environment**:  Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] Create a field. By default, it would be required. Now toggle it to hidden. Note that you would just see **Hidden** badge now. If you would go and edit the field you would still see that it's marked required, but it won't actually be required during booking. Go and make a booking without prefilling the hidden field value.
## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/features/form-builder/FormBuilder.tsx [hunk 1]
```diff
@@ -266,8 +266,7 @@ export const FormBuilder = function FormBuilder({
           {fields.map((field, index) => {
             const fieldType = FieldTypesMap[field.type];
 
-            // Hidden fields can't be required
-            const isRequired = field.required && !field.hidden;
+            const isRequired = field.required;
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilder.tsx:269]: No need to show the field as optional now. As mentioned by @Jaibles we would not be showing 'Optional' badge if field is hidden. There would just be Hidden badge.

File: packages/features/form-builder/FormBuilder.tsx [hunk 2]
```diff
@@ -306,8 +305,12 @@ export const FormBuilder = function FormBuilder({
                       {field.label || t(field.defaultLabel || "")}
                     </div>
                     <div className="flex items-center space-x-2">
-                      <Badge variant="gray">{isRequired ? "Required" : "Optional"}</Badge>
-                      {field.hidden ? <Badge variant="gray">Hidden</Badge> : null}
+                      {field.hidden ? (
+                        // Hidden field can't be required, so we don't need to show the Optional badge
+                        <Badge variant="gray">Hidden</Badge>
```
Comment (CarinaWolli) [inline packages/features/form-builder/FormBuilder.tsx:310]: Can you add translation for hidden and required?

**Review by CarinaWolli — APPROVED** [review]
LGTM!

File: packages/features/form-builder/FormBuilder.tsx (hunk 2 above)
Comment (hariombalhara) [reply packages/features/form-builder/FormBuilder.tsx]: Thanks for pointing out @CarinaWolli. Fixed these and Label and Placeholder as well.

**Review by emrysal — APPROVED** [review]

**Pull request merged**